internal/pkg/template/diff/diff.go

```diff
@@ -90,7 +91,75 @@ func isYAMLLeaf(node *yaml.Node) bool {
 }
 
 func parseSequence(from, to *yaml.Node) (map[string]*Node, error) {
-	return nil, nil
+	fromSeq, toSeq := make([]yaml.Node, len(from.Content)), make([]yaml.Node, len(to.Content))
+	if err := to.Decode(&toSeq); err != nil {
+		return nil, err
+	}
+	if err := from.Decode(&fromSeq); err != nil {
+		return nil, err
+	}
+	cachedDiff, cachedErr := make(map[string]*Node), make(map[string]error)
+	lcsIdxInFrom, lcsIdxInTo := longestCommonSubsequence(fromSeq, toSeq, func(inA, inB int) bool {
+		diff, err := parse(&(fromSeq[inA]), &(toSeq[inB]), "")
+		if diff != nil { // NOTE: cache the diff only if a modification could have happened at this position.
+			cachedDiff[cachedKey(inA, inB)] = diff
+			cachedErr[cachedKey(inA, inB)] = err
+		}
+		return err == nil && diff == nil // NOTE: we swallow the error for now.
+	})
+	childKey := seqChildKeyFunc()
+	children := make(map[string]*Node)
+	var f, t, i int
+	for {
+		if i >= len(lcsIdxInFrom) {
+			break
+		}
+		lcsF, lcsT := lcsIdxInFrom[i], lcsIdxInTo[i]
+		switch {
+		case f == lcsF && t == lcsT: // Match.
+			f++
+			t++
+			i++
+			// TODO(lou1415926): (x unchanged items)
+		case f != lcsF && t != lcsT: // Modification.
+			// TODO(lou1415926): handle list of maps modification
```

Comment:
`diff` contains a part of what we want!

I think of "list of maps" as having two scenarios: 1. the maps are "implicitly" keyed; 2. otherwise

Consider this example for "implicit" key
```yaml
- Name: Efe
  Boss: True
- Name: Boss # You can't have another person named "Boss" here! i.e. implicitly keyed.
  Boss: True

# to ⬇️ 

- Name: Efe
  Boss: True
- Name: Boss
   Boss: So True
```

I want this diff to be rendered as 

```yaml
~ - Name: Boss
  ~ Boss: True -> So True
```
That is, the "implicit key" here is `Name`. The current implementation does not handle this.

An example of "unkeyed" items:
```
- Shape: round
  Taste: sweet
- Shape: round
  Taste: sweet
  Color: orange

⬇️ 
- Shape: round
 Taste: sweet
- Shape: round
  Taste: bitter
  Color: orange
```

The `diff` should be:
```
~ - Shape: round
  ~ Taste: sweet -> bitter
  ~ Color: orange
```

instead of 
```
~ - Taste: sweet -> bitter
```